lib/dvbjs.js

```diff
@@ -131,30 +155,130 @@ dvb.find = function(searchString, callback) {
+    }).nodeify(callback);
+};
 
-        try {
-            var points = JSON.parse(data).stopFinder.points;
-        } catch (e) {
-            return callback(err);
+dvb.pins = function pins(swlat, swlng, nelat, nelng, pintypes, callback) {
+
+    var options = {
+        'url' : 'https://www.dvb.de/apps/map/pins',
+        'qs' : {
+            'showLines' : 'true',
+            'swlat' : swlat,
+            'swlng' : swlng,
+            'nelat' : nelat,
+            'nelng' : nelng,
+            'pintypes' : pintypes
         }
+    };
+    
+    return requestP(options)
+    .then(function(data) {
+        //Create an array out of the string response by splitting between each: ","
+        //Drop the first two characters: " and [ 
+        //and the last last two characters: ] and "
+        return data.slice(2, data.length-2).split('\",\"');
+        
+    })
+    .map(function(elem) {
+        return parsePin(elem, pintypes);
+    }).nodeify(callback);
+};
+
+dvb.address = function address(lat, lng, callback) {
 
-        if (!Array.isArray(points)) {
-            points = [points.point];
+    var options = {
+        'url' : 'https://www.dvb.de/apps/map/address',
+        'qs' : {
+            'lat' : lat,
+            'lng' : lng
         }
+    };
 
-        var json = points.filter(isDresden).map(convertStop);
+    return requestP(options)
+    .then(function(data) {
+        data.slice(1, data.length).split('|');
+        return {
+            city: data[0],
+            address: data[1]
+        };
+    }).nodeify(callback);
+};
+
+dvb.coords = function coords(poiID, callback) {
+   
+    var options = {
+        'url' : 'https://www.dvb.de/apps/map/coordinates',
+        'qs' : {
+            'id' : poiID,
+        }
+    };
 
-        callback(null, json);
-    });
+    return requestP(options)
+    .then(function(data) {
+        return data.slice(1, data.length-1).split('|');
+    }).nodeify(callback);
 
 };
 
+ function parsePin(dataAsString, pintypes) {
+    
+    var jsonElem;
+    if (pintypes === 'stop') {
+        jsonElem = {
+            'id' : dataAsString.split('|||')[0],
+            'name' : dataAsString.split('||')[1].split('|')[1],
+            'coords' : [
+                dataAsString.split('||')[1].split('|')[2],
+                dataAsString.split('||')[1].split('|')[3]
+            ],
+            'connections' : dataAsString.split('||')[2]   
+        };
+        return jsonElem;
+
+    } else if (pintypes === 'platform') {
+        jsonElem = {
+            'name' : dataAsString.split('||')[1].split('|')[0],
+            'coords' : [
+                dataAsString.split('||')[1].split('|')[1],
+                dataAsString.split('||')[1].split('|')[2]
+            ],
+            '?' : dataAsString.split('||')[1].split('|')[3]
+        };
+        return jsonElem;
+
+    } else if (_.contains(['poi', 'rentabike', 'ticketmachine', 'carsharing', 'parkandride'], pintypes)) {
```

Comment:
Seems to me that lodash could be something worth keeping. For now it is only used in this line of code, but i remember from previous projects that i'm using lodash a lot.